codespell_lib/data/dictionary.txt

```diff
@@ -7801,7 +7823,11 @@ cummunicate->communicate
 cumulatative->cumulative
 cumulattive->cumulative
 cuncurency->concurrency
-cunter->counter
+cunt->count, hunt,
+cunted->counted, hunted,
+cunter->counter, hunter,
+cunting->counting, hunting,
+cunts->counts, hunts,
```

Comment:
:rofl:

Reply:
Sorry, but these are perfectly valid words, the only place those corrections should possibly be is in the rare dictionary, but I wouldn't call them particularly rare, although it's likely people may mean the corrections over the word itself.

Reply:
>but I wouldn't call them particularly rare, although it's likely people may mean the corrections over the word itself.

I do too: https://github.com/SpaceVim/SpaceVim/pull/795

So you think I should just remove them entirely?  I'm fine with doing that.

Reply:
> So you think I should just remove them entirely? I'm fine with doing that.

No, your example proves they are valid typos, but because they are valid words, the CI won't allow them in the base dictionary. I'd probably suggest putting them in rare for now, and if people complain they're being forced to choose between typos and swearing, we could create a prudish dictionary where we for example gather all the swear words and their potential corrections, then people could explicitly choose whether they intend to swear or not in whatever they're checking.

We should be safe from the Scunthorpe problem, because of the way codespell works...

Reply:
@peternewman I suspect users will never want that.  In its current state, codespell spits out a lot of false positives for brand names, etc.

On a different note, @peternewman, often codespell reports false positives for short words such as `nd`, `te`, `fo`, etc.  What do you think about adding an option to skip detected typos if the length is less than or equal to a specified value, like 2?  Or, add an option to skip short (of length >=3) typos?

Reply:
> @peternewman I suspect users will never want that.

A "sanitising" dictionary, or at least words that are likely to be safe corrections if you aren't intending to swear. It's certainly not required, but would probably be an obvious way to split the rare dictionary further in future.

> In its current state, codespell spits out a lot of false positives for brand names, etc.

Can you give some examples? I suspect given the various ways brands continue to butcher real words for their names those kind of things just need disabling on a case by case basis, unless they're big multinationals like Microsoft. But we could create a brands dictionary (but perhaps leave it enabled by default), similar to names for typos that are also brands.

> On a different note, @peternewman, often codespell reports false positives for short words such as `nd`, `te`, `fo`, etc. What do you think about adding an option to skip detected typos if the length is less than or equal to a specified value, like 2? Or, add an option to skip short (of length >=3) typos?

I don't really follow, can you give some examples, probably in a new issue.